Add rrule.Set.String() and rrule.Set.Recurrences()

wangjohn opened the pull request:
These two methods allow for easier serialization of rrule sets. I've needed this in my applications for storing recurrences to the database and passing it between services. It seems like others have also needed this as well (see https://github.com/teambition/rrule-go/issues/6).

damoye commented on rruleset.go:
```diff
@@ -13,6 +15,28 @@ type Set struct {
+
+func (set *Set) Recurrences() []string {
+	res := []string{}
+	for _, o := range set.rrule {
```
'o' is little confusing? I think 'item' is better.

damoye commented on rruleset.go:
```diff
@@ -13,6 +15,28 @@ type Set struct {
+		res = append(res, fmt.Sprintf("RRULE:%v", o.String()))
+	}
+	for _, o := range set.rdate {
+		res = append(res, fmt.Sprintf("RDATE:%v", o.Format(strformat)))
```
'Z' in "20060102T150405Z" means UTC. So it should be `o.UTC().Format(strformat)`. Please use function `timeToStr`:
`res = append(res, fmt.Sprintf("RDATE:%s", timeToStr(o)))`

damoye commented on rruleset.go:
```diff
@@ -13,6 +15,28 @@ type Set struct {
+func (set *Set) Recurrences() []string {
+	res := []string{}
+	for _, o := range set.rrule {
+		res = append(res, fmt.Sprintf("RRULE:%v", o.String()))
```
`fmt.Sprintf("RRULE:%s", o)` is better.

wangjohn commented:
I've updated the PR to address your comments. I still do believe that the `set.Recurrences()` function would be nice to export, since many of the applications I have worked on actually store the recurrence rules in an array and not as a full string. Other examples of this include the Google Calendar API (https://developers.google.com/calendar/v3/reference/events#recurrence).

I'm happy to remove the exported `set.Recurrence` function, but I do believe it is helpful. Obviously, this function can be recreated using `strings.Split` on the stringified set, but I think `set.Recurrence` is somewhat cleaner.

damoye commented:
I agree with you. There are still two problems:

1. The function name should be `set.Recurrence` as same as Google Calendar API.
2. Please add comment for this exported function.

damoye commented:
I agree with you. There is still two problems:
1. The function name should be `set.Recurrence` according to Google Calendar API.
2. Please add comment for this function.

wangjohn commented:
Sorry for the long turnaround time here, but I've addressed your last two comments in b4f2fd1. Hopefully this should make the patch mergeable.

damoye commented:
Thanks a lot!
